unstructured/ingest/connector/pinecone.py

```diff
@@ -114,8 +119,19 @@ def write(self, docs: t.List[BaseIngestDoc]) -> None:
                         "id": element.pop("element_id", None),
                         "values": element.pop("embeddings", None),
                         "metadata": {
-                            k: (json.dumps(v) if isinstance(v, list) else v)
-                            for k, v in flatten_dict(element["metadata"], separator="-").items()
+                            # Since pinecone does not allow lists of any type other than str,
```

Comment:
Yeah. I had something similar in ChromaDB. I added a list flattener to `flatten_dict` and made it false by default. Maybe this way is better.

Reply:
I think it depends on whether the fields under that complex list object should be indexable or not. If those subfields of the list object should definitely be indexable, then I think your approach makes more sense. 

But if it's likely that users won't index the fields under the complex list object, then I think it's easier to retrieve the complex list object to its original form when it's a `json.dumps` string, because they can simply do `json.loads ` to retrieve the object to its original form. Whereas if it is a flattened list, it'll need some custom code to de-flatten the object

Reply:
two comments:
* @potter-potter, there's actually already a supported method to flatten dictionaries supported in the unstructured library: [base.py#L178](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/staging/base.py#L178) 
* @ahmetmeleq, given the availability of indexing capabilities supported byb pinecone, my opinion is we should flatten the entire json and preserve anything that meets these fields, as well as the original JSON being dumped as a string to a new field in the pinecone metadata so a user can  extract it out with ease.

Reply:
@rbiseck3 That's the one (base.py) I'm using in chromadb. Just had to add a flatten list to it. (and made it false by default)

Reply:
Agreed, and addressed with [make all element fields indexable, add element_serialized](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/e1a6365f7f674f57c9e3a06b3190abb33d59341c)